apps/laboratory/tests/shared/pages/ModalPage.ts

```diff
@@ -5,20 +5,25 @@ export class ModalPage {
   private readonly baseURL = BASE_URL
 
   private readonly connectButton: Locator
+  private readonly url: string
 
   constructor(
     public readonly page: Page,
-    public readonly library: string
+    public readonly library: string,
+    public readonly siwe: boolean
```

Comment:
I'd advise against boolean parameters like this. Would prefer a string union type (e.g. `"siwe" | "regular"`) or better abstraction

Reply:
decided against inheritance/polymorphism for now. when `email` comes this needs to be refactored to accommodate this all better